packages/animations/browser/src/render/renderer.ts

```diff
@@ -5,137 +5,19 @@
-
-    // cache the delegates to find out which cached delegate can
-    // be used by which cached renderer
-    const delegate = this.delegate.createRenderer(hostElement, type);
-    if (!hostElement || !type || !type.data || !type.data['animation']) {
-      let renderer: BaseAnimationRenderer|undefined = this._rendererCache.get(delegate);
-      if (!renderer) {
-        // Ensure that the renderer is removed from the cache on destroy
-        // since it may contain references to detached DOM nodes.
-        const onRendererDestroy = () => this._rendererCache.delete(delegate);
-        renderer =
-            new BaseAnimationRenderer(EMPTY_NAMESPACE_ID, delegate, this.engine, onRendererDestroy);
-        // only cache this result when the base renderer is used
-        this._rendererCache.set(delegate, renderer);
-      }
-      return renderer;
-    }
-
-    const componentId = type.id;
-    const namespaceId = type.id + '-' + this._currentId;
-    this._currentId++;
-
-    this.engine.register(namespaceId, hostElement);
-
-    const registerTrigger = (trigger: NestedAnimationTriggerMetadata) => {
-      if (Array.isArray(trigger)) {
-        trigger.forEach(registerTrigger);
-      } else {
-        this.engine.registerTrigger(componentId, namespaceId, hostElement, trigger.name, trigger);
-      }
-    };
-    const animationTriggers = type.data['animation'] as NestedAnimationTriggerMetadata[];
-    animationTriggers.forEach(registerTrigger);
-
-    return new AnimationRenderer(this, namespaceId, delegate, this.engine);
-  }
-
-  begin() {
-    this._cdRecurDepth++;
-    if (this.delegate.begin) {
-      this.delegate.begin();
-    }
-  }
-
-  private _scheduleCountTask() {
-    queueMicrotask(() => {
-      this._microtaskId++;
-    });
-  }
-
-  /** @internal */
-  scheduleListenerCallback(count: number, fn: (e: any) => any, data: any) {
-    if (count >= 0 && count < this._microtaskId) {
-      this._zone.run(() => fn(data));
-      return;
-    }
-
-    if (this._animationCallbacksBuffer.length == 0) {
-      queueMicrotask(() => {
-        this._zone.run(() => {
-          this._animationCallbacksBuffer.forEach(tuple => {
-            const [fn, data] = tuple;
-            fn(data);
-          });
-          this._animationCallbacksBuffer = [];
-        });
-      });
-    }
-
-    this._animationCallbacksBuffer.push([fn, data]);
-  }
-
-  end() {
-    this._cdRecurDepth--;
-
-    // this is to prevent animations from running twice when an inner
-    // component does CD when a parent component instead has inserted it
-    if (this._cdRecurDepth == 0) {
-      this._zone.runOutsideAngular(() => {
-        this._scheduleCountTask();
-        this.engine.flush(this._microtaskId);
-      });
-    }
-    if (this.delegate.end) {
-      this.delegate.end();
-    }
-  }
-
-  whenRenderingDone(): Promise<any> {
-    return this.engine.whenRenderingDone();
-  }
-}
+type FactoryWithListenerCallback = RendererFactory2&
```

Comment:
nit: let's include "Animation" here for readability:
```suggestion
type AnimationFactoryWithListenerCallback = RendererFactory2&
```